### PR TITLE
sqs+sns integration

### PR DESCRIPTION
**Pull request by isaev-co**
Frontend: https://d111triofks8tr.cloudfront.net/

expected score: 4/6 (no additional tasks were done 😿)

result notification via email 
Screenshot 2020-11-24 at 02 19 16

### CODE REVIEW

**Review by OHalahan — COMMENTED** [review]
joining to get a notification once PR is updated

File: product-service/serverless.yml [hunk 1]
```diff
@@ -40,3 +84,10 @@ functions:
           method: post 
           cors: true
 
+  catalogBatchProcess:
+    handler: handler.catalogBatchProcess
+    events:
+      - sqs:
+          batchSize: 1
```
Comment (OHalahan) [inline product-service/serverless.yml]: should be 5 according to task description
image

**Review by OHalahan — APPROVED** [review]

File: product-service/src/catalogBatchProcess.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      const product = JSON.parse(record.body);
+
+      const { title, artist, description, price: stringPrice, coverurl, count: stringCount } = product;
+      const price = parseInt(stringPrice);
+      const count = parseInt(stringCount);
+
+      await createProduct(title, artist, description, price, coverurl, count)
+
+      console.log(`product was added: ${JSON.stringify(product)}`);
+      await sns.publish({
```
Comment (OHalahan) [inline product-service/src/catalogBatchProcess.js]: products are sent one by one, but should be sent in batches once all products from event are processed

Comment (OHalahan) [reply product-service/src/catalogBatchProcess.js]: I mean task does not state it clearly, but that would be a logical way of processing a batch of 5 received products

File: product-service/serverless.yml (hunk 1 above)
Comment (isaev-co) [reply product-service/serverless.yml]: updated to 3 (dont want to create csv with a lot of data)

File: product-service/src/catalogBatchProcess.js (hunk 2 above)
Comment (isaev-co) [reply product-service/src/catalogBatchProcess.js]: yep, fixed (batch size updated to 3) and notification sending after whole batch processing with the corresponding email
Screenshot 2020-11-26 at 19 28 59

**Pull request merged**